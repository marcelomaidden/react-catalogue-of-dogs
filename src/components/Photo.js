import React from 'react';
import PropTypes from 'prop-types';
import '../stylesheets/Detail.css';

const Photo = ({
  id,
  url,
  photoStyle,
  photoContainer,
}) => (
  <div className={`${photoContainer} col-12`} key={id}>
    <img className={photoStyle} src={url} alt="Dog" />
  </div>
);

Photo.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  url: PropTypes.string.isRequired,
  photoStyle: PropTypes.string.isRequired,
  photoContainer: PropTypes.string.isRequired,
};

export default Photo;
