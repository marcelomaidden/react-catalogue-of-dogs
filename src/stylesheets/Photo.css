.photo-container {
  height: 250px;
}

.photo {
  object-fit: cover;
  width: 100%;
  height: 250px;
}
