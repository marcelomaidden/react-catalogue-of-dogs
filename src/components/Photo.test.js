import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import Photo from "./Photo";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("should render a Photo component", () => {
  act(() => {
    render(
      <Photo   
        id="T-0EW-SEbsE"
        url="https://images.unsplash.com/photo-1548199973-03cce0bbc87b?crop=entropy\u0026cs=tinysrgb\u0026fit=max\u0026fm=jpg\u0026ixid=Mnw1MTAzN3wwfDF8c2VhcmNofDF8fGRvZ3N8ZW58MHx8fHwxNjE1MzkxNjUx\u0026ixlib=rb-1.2.1\u0026q=80\u0026w=400"
        photoStyle="photo"
        photoContainer="photo-container"
      />, 
      container
    );
  });

  expect(
    container.innerHTML
  ).toMatchSnapshot(); 
});
