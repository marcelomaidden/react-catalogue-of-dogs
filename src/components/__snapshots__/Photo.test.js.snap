// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Photo component 1`] = `"<div class=\\"photo-container col-12\\"><img class=\\"photo\\" src=\\"https://images.unsplash.com/photo-1548199973-03cce0bbc87b?crop=entropy\\\\u0026cs=tinysrgb\\\\u0026fit=max\\\\u0026fm=jpg\\\\u0026ixid=Mnw1MTAzN3wwfDF8c2VhcmNofDF8fGRvZ3N8ZW58MHx8fHwxNjE1MzkxNjUx\\\\u0026ixlib=rb-1.2.1\\\\u0026q=80\\\\u0026w=400\\" alt=\\"Dog\\"></div>"`;
